sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -387,27 +398,43 @@ def sdc_pandas_dataframe_append(df, other, ignore_index=True, verify_integrity=F
     ty_checker.check(df, DataFrameType)
     # TODO: support other array-like types
     ty_checker.check(other, DataFrameType)
-    # TODO: support index in series from df-columns
-    if not isinstance(ignore_index, (bool, types.Boolean, types.Omitted)) and not ignore_index:
-        ty_checker.raise_exc(ignore_index, 'boolean', 'ignore_index')
 
     if not isinstance(verify_integrity, (bool, types.Boolean, types.Omitted)) and verify_integrity:
         ty_checker.raise_exc(verify_integrity, 'boolean', 'verify_integrity')
 
     if not isinstance(sort, (bool, types.Boolean, types.Omitted)) and sort is not None:
         ty_checker.raise_exc(sort, 'boolean, None', 'sort')
 
-    args = (('ignore_index', True), ('verify_integrity', False), ('sort', None))
+    if not isinstance(ignore_index, (bool, types.Boolean, types.Omitted)):
+        ty_checker.raise_exc(ignore_index, 'boolean', 'ignore_index')
+
+    none_or_numeric_indexes = ((isinstance(df.index, types.NoneType) or isinstance(df.index, types.Number))
+                               and (isinstance(other.index, types.NoneType) or isinstance(other.index, types.Number)))
+
+    indexes_comparable = check_types_comparable(df.index, other.index) or none_or_numeric_indexes
```

Comment:
You could move these lines to elif-block and use it only there if the condition is True.